benchmarks/OptimizationFrameworks/optimal_powerflow.md

```diff
@@ -0,0 +1,1767 @@
+    end
+
+    #power_balance_q_con
+    for (i,bus) in ref[:bus]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_flow_p_from_con
+    for (l,i,j) in ref[:arcs_from]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_flow_p_to_con
+    for (l,i,j) in ref[:arcs_to]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_flow_q_from_con
+    for (l,i,j) in ref[:arcs_from]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_flow_q_to_con
+    for (l,i,j) in ref[:arcs_to]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_flow_vad_con
+    for (l,i,j) in ref[:arcs_from]
+        branch = ref[:branch][l]
+        push!(con_lbs, branch["angmin"])
+        push!(con_ubs, branch["angmax"])
+    end
+
+    #power_flow_mva_from_con
+    for (l,i,j) in ref[:arcs_from]
+        branch = ref[:branch][l]
+        push!(con_lbs, -Inf)
+        push!(con_ubs, branch["rate_a"]^2)
+    end
+
+    #power_flow_mva_to_con
+    for (l,i,j) in ref[:arcs_to]
+        branch = ref[:branch][l]
+        push!(con_lbs, -Inf)
+        push!(con_ubs, branch["rate_a"]^2)
+    end
+
+    model_variables = length(var_init)
+    model_constraints = length(opf_constraints!(similar(con_lbs), var_init))
+    println("variables: $(model_variables), $(length(var_lb)), $(length(var_ub))")
+    println("constraints: $(model_constraints), $(length(con_lbs)), $(length(con_ubs))")
+
+    nlp = ADNLPModels.ADNLPModel!(opf_objective, var_init, var_lb, var_ub, opf_constraints!, con_lbs, con_ubs, backend = :optimized)
+
+    model_build_time = time() - time_model_start
+
+
+    time_solve_start = time()
+
+    output = NLPModelsIpopt.ipopt(nlp)
+    cost = output.objective
+    feasible = (output.primal_feas <= 1e-6)
+    
+    solve_time = time() - time_solve_start
+    total_time = time() - time_data_start
+
+
+    println("")
+    println("\033[1mSummary\033[0m")
+    println("   case........: $(file_name)")
+    println("   variables...: $(model_variables)")
+    println("   constraints.: $(model_constraints)")
+    println("   feasible....: $(feasible)")
+    println("   cost........: $(round(Int, cost))")
+    println("   total time..: $(total_time)")
+    println("     data time.: $(data_load_time)")
+    println("     build time: $(model_build_time)")
+    println("     solve time: $(solve_time)")
+    # println("      callbacks: $(total_callback_time)")
+    println("")
+
+    return Dict(
+        "case" => file_name,
+        "variables" => model_variables,
+        "constraints" => model_constraints,
+        "feasible" => feasible,
+        "cost" => cost,
+        "time_total" => total_time,
+        "time_data" => data_load_time,
+        "time_build" => model_build_time,
+        "time_solve" => solve_time,
+        #"time_callbacks" => TBD,
+    )
+end
+
+solve_opf_nlpmodels("data/pglib_opf_case5_pjm.m")
+```
+
+## Nonconvex
```

Comment:
I think instead of Nonconvex it should have the NLPModels implementation which seems to be quite performant now.